tests/eas/generic.py

```diff
@@ -252,9 +256,80 @@ def exp_power(row):
             columns = power.keys()
             return pd.Series([power[c] for c in columns], index=columns)
+
+        self.plot_expected_util(experiment)
+
         return self._sort_power_df_columns(
             task_utils_df.apply(exp_power, axis=1))
 
+    def plot_expected_util(self, experiment):
+        """
+        Create a plot of the expected per-CPU utilization for the experiment
+        The plot is then outputted to the test results directory.
+
+        :param experiment: The :class:Experiment to examine
+        :returns: A Pandas DataFrame with a column each node in the energy model
+                  (keyed with a tuple of the CPUs contained by that node) and a
+                  "power" column with the sum of other columns. Shows the
+                  estimated *optimal* power over time.
+        """
+        task_utils_df = self.get_task_utils_df(experiment)
+        nrg_model = self.te.nrg_model
+        trace = self.get_trace(experiment)
+
+        cpus = nrg_model.cpus
+        data = []
+        index = []
+        for time, utils_row in task_utils_df.iterrows():
+            task_utils = utils_row.to_dict()
+            expected_utils = nrg_model.get_optimal_placements(task_utils)[0]
+
+            row = []
+            for cpu, util in enumerate(expected_utils):
+                row.append(util)
+
+            data.append(row)
+            index.append(time)
+
+        df = pd.DataFrame(data, index=index)
+        fig, ax = plt.subplots(len(cpus), 1, figsize=(16, 1.8 * len(cpus)))
+
+        # Check if big.LITTLE data is available for a more detailled plot
+        if trace.has_big_little:
+            little_cap = self.te.platform['nrg_model']['little']['cpu']['cap_max']
+            big_cpus = self.te.platform['clusters']['big']
+            little_cpus = self.te.platform['clusters']['little']
+
+        for cpu in cpus:
+            tdf = df[cpu]
+
+            # big.LITTLE-specific beautifying
+            if trace.has_big_little and cpu in little_cpus:
+                ax[cpu].set_ylim((0, little_cap))
+                color = 'blue'
+            else:
+                ax[cpu].set_ylim((0, 1024))
+                color = 'red'
+
+            tdf.plot(ax=ax[cpu], drawstyle='steps-post', title="CPU{}".format(cpu), color=color)
+            ax[cpu].fill_between(tdf.index, tdf, 0, step='post', color=color)
+            ax[cpu].set_ylabel('Utilization')
+
+            # Grey-out areas where utilization == 0
+            ffill = False
+            prev = 0.0
+            for time, util in tdf.iteritems():
+                if ffill:
+                    ax[cpu].axvspan(prev, time, facecolor='gray', alpha=0.1, linewidth=0.0)
+                    ffill = False
+                if util == 0.0:
+                    ffill = True
+
+                prev = time
+
+        figname = os.path.join(experiment.out_dir, 'expected_placement.png')
+        pl.savefig(figname, bbox_inches='tight')
+
```

Comment:
Does this also show the plot inline when you call it from a notebook?

Reply:
I'm not super knowledgeable about executor/experiments, so I only tested this by running generics.py. I suppose it could be used in a notebook by feeding it experiments from an executor, but I haven't tried this. Is this what you meant ?

Reply:
Yeah. Not important I was just curious.